frontend/src/components/App/app.jsx

```diff
@@ -89,14 +89,10 @@ function App() {
                         </Route>
 
                         {/* Profile */}
-                        <Route path="profile">
+                        <Route path="profile/:uid" element={<ProfilePage />} >
 
-                            {/* Company profile */}
-                            <Route path="company/:uid" element={<ProfilePage />}>
-
-                                {/* Profile editing */}
-                                <Route path="edit" element={<ProfileEditing />} />
-                            </Route>
+                            {/* Profile editing */}
+                            <Route path="edit" element={<ProfileEditing />} />
```

Comment:
The nested route for profile editing may not work as expected since "profile/:uid" is not an outlet parent. Ensure <ProfilePage /> renders an <Outlet /> for nested routes.
```
// In ProfilePage component:
import { Outlet } from "react-router-dom";
return (
  <div>
    {/* profile content */}
    <Outlet />
  </div>
);
```